### PR TITLE
Actually register modeled TE header in header parsing infra

### PR DESCRIPTION
**Pull request by raboof**
And add tests. Follow-up to #3616

### CODE REVIEW

**Review by johanandren — APPROVED** [review]

**Review by jrudolph — APPROVED** [review]
LGTM, do we already have a javadsl model for the TE header?

**Comment by raboof** [comment]
> do we already have a javadsl model for the TE header?

Good point, added

File: akka-http-core/src/main/java/akka/http/javadsl/model/headers/TE.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ */
+
+package akka.http.javadsl.model.headers;
+
+/**
+ *  Model for the `Transfer-Encoding` header.
+ *  Specification: http://tools.ietf.org/html/draft-ietf-httpbis-p1-messaging-26#section-3.3.1
+ */
+public abstract class TE extends akka.http.scaladsl.model.HttpHeader {
+    public abstract Iterable<akka.http.javadsl.model.TransferEncoding> getAcceptedEncodings();
```
Comment (jrudolph) [inline akka-http-core/src/main/java/akka/http/javadsl/model/headers/TE.java:12]: The implementation seem to be missing?

Comment (raboof) [reply akka-http-core/src/main/java/akka/http/javadsl/model/headers/TE.java]: doh, I'm clearly trying to do too much stuff in parallel here..

**Pull request merged**